### PR TITLE
docs: add documentation for relative asset URLs

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

**Comment by IgorMinar** [comment]
rendered preview: https://github.com/matsko/angular/blob/component_assets_guide/modules/angular2/docs/core/13_relative_assets.md

File: modules/angular2/docs/core/13_relative_assets.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+export class MyCmp {
+}
+```
+
+The default value for the package scheme can be changed by overriding the `PACKAGE_ROOT_URL`
+token when bootstrapping an Angular2 application:
+
+```ts
+import {MyCmp} from "app/my_cmp";
+import {PACKAGE_ROOT_URL} from "angular2/compiler";
```
Comment (gkalpak) [inline modules/angular2/docs/core/13_relative_assets.md]: This seems to be exported by `angular2/core`.

File: modules/angular2/docs/core/13_relative_assets.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import {bootstrap} from "angular2/platform/browser";
+import {provide} from "angular2/core";
+
+bootstrap(MyCmp, [
+  provide(PACKAGE_ROOT_URL, { toValue: "/app/assets" })
```
Comment (gkalpak) [inline modules/angular2/docs/core/13_relative_assets.md]: _toValue_  -->  _useValue_ (?)
<sub>(although `foo` seems to work just fine as well :stuck_out_tongue:)</sub>

**Comment by mhevery** [comment]
@matsko / @IgorMinar What is the status of this PR?

**Comment by IgorMinar** [comment]
I believe that @wardbell incorporated this into a dev guide. Ward, can you please confirm?

if not can you please make sure that this happens?

There are two pieces of information that are now missing from this PR: info about __moduleName in SystemJS and I that `package:` prefix is currently incompatible with the offline template compiler.

We should also notice that in order to write code that works both with runtime and offline compiler it's best to use moduleId and relative paths. This ensure that the code works in both modes.

The runtime compiler uses both moduleId and relative path, while offline compiler uses only the relative path and does similar path resolution as we do with moduleId.

**Comment by wardbell** [comment]
Actually, I only incorporated the `moduleId` option and only with respect to the commonjs format, `module.id`.

See the [Component-relative Paths](https://angular.io/docs/ts/latest/cookbook/component-relative-paths.html) cookbook.

We _used_ to describe the `__moduleName` option. I removed that from the docs when we made our general commitment to commonjs module format.

I did not describe the `package` option because (a) we have never encouraged it and (b) it didn't seem nearly as useful. To truly support the goal of moving modules around would require multiple provisions of `provide(PACKAGE_ROOT_URL, { toValue: "..." })` at every non-leaf-level of the component tree.  That just isn't much fun.

I recommend that 
1. We close this PR _with regret_
1. Deprecate the `package` option

**Comment by mhevery** [comment]
OK, closing

**Pull request closed without merging**